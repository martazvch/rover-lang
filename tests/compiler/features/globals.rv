var a = 15.6
var number = 15.6 + .6

var name = "Blender"

number = 90
name = "Best software"

fn main() {}