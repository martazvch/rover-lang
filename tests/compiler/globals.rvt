// Tests literals

config:
ignore main-call

code:
var a = 15.6
fn main() {}

expect:
-- Function main
OP_NULL
OP_RETURN

OP_CONSTANT index 0, value 15.6
OP_DEFINE_GLOBAL index 0

================

code:
var number = 15.6 + .6
fn main() {}

expect:
-- Function main
OP_NULL
OP_RETURN

OP_CONSTANT index 0, value 15.6
OP_CONSTANT index 1, value 0.6
OP_ADD_FLOAT
OP_DEFINE_GLOBAL index 0

================

code:
var number = 15.6 + .6
var name = "Blender"

number = 90
name = "Best software"
fn main() {}

expect:
-- Function main
OP_NULL
OP_RETURN

OP_CONSTANT index 0, value 15.6
OP_CONSTANT index 1, value 0.6
OP_ADD_FLOAT
OP_DEFINE_GLOBAL index 0
OP_CONSTANT index 2, value "Blender"
OP_DEFINE_GLOBAL index 1
OP_CONSTANT index 3, value 90
OP_CAST_TO_FLOAT
OP_SET_GLOBAL index 0
OP_CONSTANT index 4, value "Best software"
OP_SET_GLOBAL index 1

================

