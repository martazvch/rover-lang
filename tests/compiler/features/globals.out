-- main --
ret_naked

-- global scope --
constant index 0, value 15.6
OP_DEFINE_GLOBAL index 0

constant index 1, value 15.6
constant index 2, value 0.6
OP_ADD_FLOAT
OP_DEFINE_GLOBAL index 1

constant index 3, value Blender
OP_DEFINE_GLOBAL index 2

constant index 4, value 90
OP_CAST_TO_FLOAT
OP_SET_GLOBAL index 1

constant index 5, value Best software
OP_SET_GLOBAL index 2

OP_GET_GLOBAL index 3, <fn main>
call index 0
